eclipse/src/saros/editor/internal/ContributionAnnotationManager.java

```diff
@@ -321,8 +370,20 @@ private void replaceInHistory(
             + oldAnnotation.getSource());
   }
 
+  /**
+   * Removes all annotations from all annotation models that are currently stored in the history of
+   * all users.
+   */
   private void removeAllAnnotations() {
-    for (Queue<ContributionAnnotation> queue : sourceToHistory.values())
-      while (!queue.isEmpty()) removeFromHistory(queue.peek());
+
+    final Set<IAnnotationModel> annotationModels = new HashSet<>();
+
+    for (final History history : sourceToHistory.values())
+      while (!history.entries.isEmpty())
```

Comment:
Is this an eternal loop? The def. of `peek` says `Retrieves, but does not remove, the head (first element) of this list.`.
Why not simply `for (HistoryEntry entry : history.entries)` ?

Reply:
THX !!!

Reply:
maybe `for (HistoryEntry entry : history.entries.sublist(0, history.entries.size())` if your intention is to clear `history.entries` to avoid that for loop altogether.